### ISSUE TITLE
[false-positive] 'type unsafe comparison' on DateTime objects

### ISSUE THREAD

**Issue by b1rdex** [kalessil/phpinspectionsea#19]
```php
$date1 = '2016-11-02';
$date2 = '2016-11-02';
(new DateTime($date1)) == (new DateTime($date2))
```

You can't check DateTime equality, but php allows them to compare using ==, > and <.

**Comment by dryabov** [comment]
I've checked php sources, there are special comparators for `Closure`, `DateTime`, `DateTimeImmutable`, `IntlBreakIterator`, `IntlTimeZone`, `PDO`, `PDOStatement`, `ArrayObject`, `SplObjectStorage` (search for `.compare_objects` string).

**Comment by dryabov** [comment]
And most likely classes extending above ones have the same comparators (but it should be checked, seems it is not explained in docs).

**Comment by kalessil** [comment]
@dryabov : huge thanks for your research, PHP is keeping to surprise me.

**Comment by Rarst** [comment]
I've noticed "type unsafe comparison" showing up on DateTime objects as of right now, had this regressed since or something?

**Comment by kalessil** [comment]
@Rarst: would you please create a new issue for your case?
